src/lib/lifeCycle.ts

```diff
@@ -50,11 +50,12 @@ import {
 } from "./paths.js";
 import { confirm } from "./prompt.js";
 
-const ensureUserConfig = async (): Promise<UserConfig> => {
-  const userConfig = await initUserConfig();
+const ensureUserConfig = async (): Promise<void> => {
+  const maybeUserConfig = await initUserConfig();
 
-  if (userConfig !== null) {
-    return userConfig;
+  if (maybeUserConfig !== null) {
+    setUserConfig(maybeUserConfig);
```

Comment:
It works because before each command execution we do init. This is ok, but not testable. For testability it would be better to figure out a good pattern for dependency injection here, which I will definitly will have to do when I finnaly get to work on testing